### ISSUE TITLE
-fportable and linker failure

### ISSUE THREAD

**Issue by mietek** [valderman/haste-compiler#255]
Compiling `haste-compiler-0.4.3` with `-fportable` results in the following linker failure:

```
Linking dist/build/hastec/hastec ...
/app/.halcyon/ghc/lib/ghc-7.8.3/rts-1.0/libHSrts.a(Linker.o): In function `foreignExportStablePtr':
(.text+0x25b): warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: cannot find -lgmp
/app/.halcyon/ghc/lib/ghc-7.8.3/unix-2.7.0.1/libHSunix-2.7.0.1.a(HsUnix.o): In function `__hsunix_getpwent':
(.text+0xb1): warning: Using 'getpwent' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/app/.halcyon/ghc/lib/ghc-7.8.3/unix-2.7.0.1/libHSunix-2.7.0.1.a(HsUnix.o): In function `__hsunix_getpwnam_r':
(.text+0xc1): warning: Using 'getpwnam_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/app/.halcyon/ghc/lib/ghc-7.8.3/unix-2.7.0.1/libHSunix-2.7.0.1.a(HsUnix.o): In function `__hsunix_getpwuid_r':
(.text+0xd1): warning: Using 'getpwuid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
collect2: error: ld returned 1 exit status
```

Compiling without this flag succeeds.

`libgmp` is available:

```
$ env | grep LIBRARY_PATH
LIBRARY_PATH=/app/.halcyon/ghc/lib:
LD_LIBRARY_PATH=/app/.halcyon/ghc/lib:

$ ls -l /app/.halcyon/ghc/lib | grep libgmp
lrwxrwxrwx  1 root root   38 Oct 26 08:40 libgmp.so -> /usr/lib/x86_64-linux-gnu/libgmp.so.10
lrwxrwxrwx  1 root root   38 Oct 26 08:40 libgmp.so.10 -> /usr/lib/x86_64-linux-gnu/libgmp.so.10
lrwxrwxrwx  1 root root   38 Oct 31 20:53 libgmp.so.3 -> /usr/lib/x86_64-linux-gnu/libgmp.so.10
```

This is on Ubuntu 14.04 LTS, with GHC 7.8.3.
